### PR TITLE
Add Base32 extensions and unit tests for it (EXPOSUREAPP-5329)

### PR DESCRIPTION
**Pull request by mtwalli**
- Create Base32 wrapper and add unit tests for it to verify its correctness 
- Ticket: https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5329 
- Sample test data : https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/event-registration/docs/spec/event-registration-client.md#base32-examples

### CODE REVIEW

**Comment by harambasicluka** [comment]
This has to go in the feature branch for the event registration. 
ping @d4rken

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/Base32Encoder.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package de.rki.coronawarnapp.eventregistration
+
+import com.google.common.io.BaseEncoding
+
+/**
+ * Base32 Encoding wrapper around [BaseEncoding.base32]
+ */
+class Base32Encoder {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/Base32Encoder.kt:8]: What do you think about making this an extension function and using the same pattern als `decodeBase64()` / `base64()`, see `ByteString`.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/eventregistration/Base32EncoderTest.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package de.rki.coronawarnapp.eventregistration
+
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+
+class Base32EncoderTest {
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/eventregistration/Base32EncoderTest.kt:7]: Please extend `BaseTest`.

**Comment by mtwalli** [comment]
Tested on Samsung S8 Android 9, it flickers sometimes.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/Base32.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package de.rki.coronawarnapp.eventregistration
+
+import com.google.common.io.BaseEncoding
+
+/**
+ * Decodes input String into byte array using Base32 decoder
+ * @return [ByteArray]
+ */
+fun String.decodeBase32ToArray(): ByteArray = BaseEncoding.base32().decode(this)
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/Base32.kt:9]: What do you think about decoding to `ByteString` (`decodeBase64()` also uses it), less custom code from our end, and the API consumer of these extensions has more options.

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/Base32.kt:9]: `decodeBase32(): ByteString`

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/eventregistration/Base32Test.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package de.rki.coronawarnapp.eventregistration
+
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import testhelpers.BaseTest
+
+class Base32Test : BaseTest() {
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/eventregistration/Base32Test.kt:8]: Nice tests :heart:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/Base32.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/Base32.kt:11]: Good idea, Added a new variant for it. we can use what suits the use case.
`decodeBase32ToByteString`

**Review by d4rken — APPROVED** [review]
:heart_eyes:  so nice!

**Comment by mlenkeit** [comment]
Looks promising 👍

**Review by ralfgehrer — APPROVED** [review]

**Pull request merged**